code/messaging/src/phone/java/com/adobe/marketing/mobile/Messaging.java

```diff
@@ -307,15 +308,19 @@ public void call(final Event event) {
                         return;
                     }
 
+                    Map<Surface, List<Proposition>> propositions = new HashMap<>();
                     Surface surface = null;
                     for (final Map<String, Object> propositionMap : retrievedPropositions) {
+
                         final Proposition proposition = Proposition.fromEventData(propositionMap);
-                        surface = Surface.fromUriString(proposition.getScope());
-                        propositions.add(proposition);
+                        if (proposition != null) {
+                            surface = Surface.fromUriString(proposition.getScope());
+                            propositions = MessagingUtils.updateMapForSurface(surface, proposition, propositions);
+                        }
                     }
 
                     if (surface != null) {
-                        requestedPropositions.put(surface, propositions);
+                        requestedPropositions.put(surface, propositions.get(surface));
```

Comment:
Im not able to understand why are we adding a single surface entry in requestedPropositions. Am I missing anything here? Ideally all your propositions are accumulated against surface in the propositions list defined on L311.